### PR TITLE
Fix intermittent GammaRay rpi-qtauto build failures

### PR DESCRIPTION
**Pull request by martin-ejdestig**

### CODE REVIEW

**Comment by rpannek** [comment]
The only thing I'm not quite liking is that it's a temporary fix but it's not in the recipes-temporary-patches but I guess that is because it's dependent on the b2qt layer?

**Comment by martin-ejdestig** [comment]
@rpannek I did not know about recipes-temporary-patches. Makes sense, I think. More clear that it is temporary.

**Comment by kanavin** [comment]
Generally, bbappend files are hacks with poor maintainability, and I would discourage them. Make every effort to fix the original recipes please :)

**Comment by martin-ejdestig** [comment]
@kanavin Yes, I agree. Not sure how to solve it 100% correctly though (probably want to use CMAKE_SYSROOT, and that may include changes to whatever component creates the CMake toolchain path... not sure), hence the upstream bug. I would expect the real fix to be very similar to this though. In the mean time, this fixes a stability issue in our CI/CD system.

**Comment by kanavin** [comment]
Can you provide a snippet of what is problematic in the original cmake files, and how sed removes the problem please? I might be able to suggest smth better.

**Comment by martin-ejdestig** [comment]
@rpannek Moved to recipes-temporary-patches.

@sashko Not a fan (less room for actual meaningful text in short message). Especially in a small repo like meta-pelux where there really are no "components" to speak of... but changed.

**Comment by martin-ejdestig** [comment]
@kanavin 
```shell
$ cd tmp/work/cortexa7hf-neon-vfpv4-pelux-linux-gnueabi/qtbase/5.10.1+gitAUTOINC+6c6ace9d23-r0
$
$ grep -r recipe-sysroot image/usr/lib/cmake/
image/usr/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake:_qt5gui_find_extra_libs(EGL "EGL;GLESv2;bcm_host;vchostif;bcm_host;vcos;vchiq_arm" "" "/home/martin/Source/pelux/build-rpi-qtauto/tmp/work/cortexa7hf-neon-vfpv4-pelux-linux-gnueabi/qtbase/5.10.1+gitAUTOINC+6c6ace9d23-r0/recipe-sysroot/usr/include/interface/vmcs_host/linux;/home/martin/Source/pelux/build-rpi-qtauto/tmp/work/cortexa7hf-neon-vfpv4-pelux-linux-gnueabi/qtbase/5.10.1+gitAUTOINC+6c6ace9d23-r0/recipe-sysroot/usr/include/interface/vcos/pthreads;/home/martin/Source/pelux/build-rpi-qtauto/tmp/work/cortexa7hf-neon-vfpv4-pelux-linux-gnueabi/qtbase/5.10.1+gitAUTOINC+6c6ace9d23-r0/recipe-sysroot/usr/include/interface/vmcs_host/linux;/home/martin/Source/pelux/build-rpi-qtauto/tmp/work/cortexa7hf-neon-vfpv4-pelux-linux-gnueabi/qtbase/5.10.1+gitAUTOINC+6c6ace9d23-r0/recipe-sysroot/usr/include/interface/vcos/pthreads")
image/usr/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake:_qt5gui_find_extra_libs(OPENGL "GLESv2;bcm_host;vcos;vchiq_arm" "" "/home/martin/Source/pelux/build-rpi-qtauto/tmp/work/cortexa7hf-neon-vfpv4-pelux-linux-gnueabi/qtbase/5.10.1+gitAUTOINC+6c6ace9d23-r0/recipe-sysroot/usr/include/interface/vmcs_host/linux;/home/martin/Source/pelux/build-rpi-qtauto/tmp/work/cortexa7hf-neon-vfpv4-pelux-linux-gnueabi/qtbase/5.10.1+gitAUTOINC+6c6ace9d23-r0/recipe-sysroot/usr/include/interface/vcos/pthreads")
$
$ <checkout this branch and run: bitbake -c install -f qtbase>
$
$ grep -r recipe-sysroot image/usr/lib/cmake/
$ grep -r ^_qt5gui_find_extra_libs image/usr/lib/cmake/
image/usr/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake:_qt5gui_find_extra_libs(EGL "EGL;GLESv2;bcm_host;vchostif;bcm_host;vcos;vchiq_arm" "" "$ENV{PKG_CONFIG_SYSROOT_DIR}/usr/include/interface/vmcs_host/linux;$ENV{PKG_CONFIG_SYSROOT_DIR}/usr/include/interface/vcos/pthreads;$ENV{PKG_CONFIG_SYSROOT_DIR}/usr/include/interface/vmcs_host/linux;$ENV{PKG_CONFIG_SYSROOT_DIR}/usr/include/interface/vcos/pthreads")
image/usr/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake:_qt5gui_find_extra_libs(OPENGL "GLESv2;bcm_host;vcos;vchiq_arm" "" "$ENV{PKG_CONFIG_SYSROOT_DIR}/usr/include/interface/vmcs_host/linux;$ENV{PKG_CONFIG_SYSROOT_DIR}/usr/include/interface/vcos/pthreads")
```

That is, no longer any hardcoded paths to qtbase's recipe-sysroot in Qt5GuiConfigExtras.cmake.

**Comment by kanavin** [comment]
@martin-ejdestig thanks, I see. It's not an upstream cmake or yocto issue, but rather how qt5 is forming those files during its own build. Or so it seems. Probably should be even fixed in qt5 upstream?

**Comment by martin-ejdestig** [comment]
@kanavin Well, the fact that the sysroot is moved in Yocto is worked around all over the place in recipes in ways similar to this (e.g. sed it out to "" in *.pc pkg-config files and in Qt *.prl files). Maybe Qt5GuiConfigExtras.cmake.in can be changed in upstream Qt to do the correct thing. Unsure.

**Comment by kanavin** [comment]
This is the original upstream file:
https://github.com/qt/qtbase/blob/5.11/src/gui/Qt5GuiConfigExtras.cmake.in
To me, it seems like CMAKE_INCLUDE_DIR_IS_ABSOLUTE needs to be true?

**Comment by martin-ejdestig** [comment]
CMAKE_INCLUDE_DIR_IS_ABSOLUTE is only used if CMAKE_ANGLE_EGL_DLL_RELEASE is set (Windows only, I assume).

No indentation, so hard to follow. But relevant parts for us is after !!ELSE at line 53, if I am not mistaken.

**Comment by kanavin** [comment]
Definitely something for meta-qt5 people to look into. I guess the discussion is best continued there.
